rasa/core/interpreter.py

```diff
@@ -296,6 +300,96 @@ def _load_interpreter(self) -> None:
 
         self.interpreter = Interpreter.load(self.model_directory)
 
+MODEL_NAME_PREFIX = "nlu"
+
+class RasaCoreInterpreter(NaturalLanguageInterpreter):
+    def __init__(self,):
+        self._load_interpreter()
+
+    def parse(self, text: Text):
+        result = self.interpreter.parse(text, only_output_properties=False)
+        return result
+
+    def _load_interpreter(self) -> None:
+        from rasa.nlu import config
+
+        config = config.load()
+        self.trainer = Trainer(config)
+        self.interpreter = Interpreter(self.trainer.pipeline, {})
+
+    def train(self, data: TrainingData, path, **kwargs: Any) -> "Interpreter":
+        """Trains the underlying pipeline using the provided training data."""
+
+        self.training_data = data
+
+        self.training_data.validate()
+
+        context = kwargs
+
+        for component in self.trainer.pipeline:
+            updates = component.provide_context()
+            if updates:
+                context.update(updates)
+
+        # data gets modified internally during the training - hence the copy
+        # working_data = copy.deepcopy(data)
+
+        for i, component in enumerate(self.trainer.pipeline):
+            logger.info(f"Starting to train component {component.name}")
+            component.prepare_partial_processing(self.trainer.pipeline[:i], context)
+            updates = component.train(data, self.trainer.config, **context)
+            logger.info("Finished training component.")
+            if updates:
+                context.update(updates)
+        self.persist(path)
+
+        return Interpreter(self.trainer.pipeline, context)
+
+    @staticmethod
+    def _file_name(index: int, name: Text) -> Text:
+        return f"component_{index}_{name}"
+
+    def persist(
```

Comment:
don't we already have `persist` for nlu pipeline?

Reply:
it's a copy of that one but made so that it doesn't add the date and time to the folder name so that we can easily load it afterwards. I'm sure it will change later.

Reply:
But I don't see load method